src/main/angular/app/services/graph-processor.service.ts

```diff
@@ -0,0 +1,275 @@
+import {Injectable, SimpleChanges} from '@angular/core';
+import {ProjectGraphData} from '../../model/project-graph-data';
+import {Recommendation} from '../../model/recommendation';
+import {Project} from '../../model/project';
+import {SimpleChange} from '@angular/core';
+import {DateUtilitiesService} from './date-utilities.service';
+import {GraphProperties, Columns, Row} from '../../model/types';
+import {DataService} from './data.service';
+
+/** Provides methods to convert data to the format used by Google Charts. */
+@Injectable()
+export class GraphProcessorService {
+  constructor(private dateUtilities: DateUtilitiesService) {}
+
+  /** Initialize the chart properties with empty data. */
+  initProperties(): GraphProperties {
+    const options: google.visualization.LineChartOptions = {
+      animation: {
+        duration: 500,
+        easing: 'out',
+        startup: true,
+      },
+      legend: {position: 'none'},
+      hAxis: {
+        gridlines: {
+          color: 'white',
+        },
+        format: 'M/d',
+        minTextSpacing: 100,
+      },
+      vAxis: {
+        minorGridlines: {
+          color: 'white',
+        },
+      },
+      series: {},
+    };
+    return {
+      options: options,
+      graphData: [],
+      columns: ['Time'],
+      type: 'LineChart',
+      title: 'Empty Graph',
```

Comment:
Does this get shown somewhere?

Reply:
Nope - it's just there so TypeScript doesn't get angry.